src/Microsoft.SqlTools.ServiceLayer/LanguageServices/AutoCompleteHelper.cs

```diff
@@ -680,5 +680,79 @@ private static string GetCompletionItemInsertName(Declaration autoCompleteItem)
                 return null;
             }
         }
+
+        /// <summary>
+        /// Converts a SQL Parser List of MethodHelpText objects into a VS Code SignatureHelp object
+        /// </summary>
+        internal static SignatureHelp ConvertMethodHelpTextListToSignatureHelp(List<Babel.MethodHelpText> methods, Babel.MethodNameAndParamLocations locations, int line, int column)
+        {
+            SignatureHelp help = new SignatureHelp();
+
+            help.Signatures = methods.Select(method =>
+            {
+                return new SignatureInformation()
+                {
+                    // Signature label format: <name> param1, param2, ..., paramn RETURNS <type>
+                    Label = method.Name + " " + method.Parameters.Select(parameter => parameter.Display).Aggregate((l, r) => l + "," + r) + " " + method.Type,
+                    Documentation = method.Description,
+                    Parameters = method.Parameters.Select(parameter => 
+                    {
+                        return new ParameterInformation()
+                        {
+                            Label = parameter.Display,
+                            Documentation = parameter.Description
+                        };
+                    }).ToArray()
+                };
+            }).ToArray();
+
+            if (help.Signatures.Length == 0)
+            {
+                return null;
+            }
+
+            // Find the matching method signature at the cursor's location
+            int signatureIndex = methods.FindIndex(method => method.Name.Contains(locations.Name));
```

Comment:
>locations [](start = 82, length = 9)

Should this be earlier? Otherwise don't we convert the methods to signatures unnecessarily? Probably not a huge perf concern, it's just that ideally we validate early exit conditions first, then do business logic. Not a blocker.